File: src/types/system.ts
export type ServiceStatus = "checking" | "ready" | "unavailable";
export interface HealthResponse {
  readonly status: "ok"; readonly service: string; readonly api_version: string;
  readonly database: { readonly status: "ready"; readonly dialect: "sqlite"; readonly schema_version: string; };
  readonly pdf_engine: "weasyprint";
}

File: backend/schemas/system.py
from typing import Literal
from pydantic import BaseModel, ConfigDict
class ApiSchema(BaseModel):
    model_config = ConfigDict(extra="forbid")
class DatabaseStatus(ApiSchema):
    status: Literal["ready"]
    dialect: Literal["sqlite"]
    schema_version: str
class HealthResponse(ApiSchema):
    status: Literal["ok"]
    service: str
    app_version: str
    api_version: str
    database: DatabaseStatus
    pdf_engine: Literal["weasyprint"]
